packages/devextreme/js/__internal/ui/m_multi_view.ts

```diff
@@ -142,14 +142,51 @@ const MultiView = CollectionWidget.inherit({
     this._initSwipeable();
   },
 
+  _ensureSelectedItemIsVisible(): void {
+    const { items } = this.option();
+    let currentSelectedIndex = this.option('selectedIndex');
+
+    if (currentSelectedIndex >= items.length) {
+      currentSelectedIndex = items.length - 1;
+      this.option('selectedIndex', currentSelectedIndex);
+    }
+
+    if (this._isItemVisible(currentSelectedIndex)) {
+      return;
+    }
+
+    const allItemsHidden = items.every((_, index) => !this._isItemVisible(index));
+    const isIndexOverflowed = (this._itemsCount() - 1) < currentSelectedIndex;
+    const canCheckItemsAhead = items.slice(currentSelectedIndex + 1)
```

Comment:
So, now we check if there is a visible item to the right twice?
I believe we can omit this check